forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -1551,6 +1564,26 @@ private boolean stateBasedAction704_attach(Card c, CardCollection unAttachList)
         return checkAgain;
     }
 
+    private boolean stateBasedAction704_5u(Player p) {
+        boolean checkAgain = false;
+
+        CardCollection toAssign = new CardCollection();
+
+        for (final Card c : p.getCreaturesInPlay().threadSafeIterable()) {
+            if (!c.hasSector()) {
+                toAssign.add(c);
+                checkAgain = true;
+            }
+        }
+
+        for (Card assignee : toAssign) { // probably would be nice for players to pick order of assigning?
+            assignee.assignSector(p.getController().chooseSector(assignee, "Assign"));
+            toAssign.remove(assignee);
```

Comment:
You should not modify the data while iterating over it

Reply:
Ha ha just found that out the hard way